src/Components/test/Ignitor/RenderBatchReader.cs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Ignitor
+{
+    public static class RenderBatchReader
```

Comment:
We have a set of unit tests that cover our `RenderBatchWriter`. If you want to have unit test coverage for `RenderBatchReader` as well, you could go to those unit tests and append some assertions that the output from the writer can be fed back into your reader and it recovers the original data.

Reply:
> We have a set of unit tests that cover our `RenderBatchWriter`. If you want to have unit test coverage for `RenderBatchReader` as well, you could go to those unit tests and append some assertions that the output from the writer can be fed back into your reader and it recovers the original data.

Great suggestion.

> Assuming we're going to keep and maintain this code, I think it would be very valuable to have unit test coverage for it. We're effectively duplicating a bunch of knowledge here, so whenever we add or change to the format of frames/edits or the serialization format, we have to remember to come and make the equivalent changes. Things could easily drift apart without sufficiently detailed checks.

Ya I totally agree. I'm going to take a quick look at those unit tests and see if it's easy to bring over. Going to time box that to 30 minutes, if it takes more we'll have to revisit so I can get all my preview7 work done 😄